### PR TITLE
initial CHANGE_LOG for 0.55

### PR DESCRIPTION
**Pull request by esc**
As title.

### CODE REVIEW

**Comment by esc** [comment]
Prose taken from: https://github.com/stuartarchibald/numba/commit/2d8b802b7a6b83407f96d10fa926f5b4f168da2f 

Thank you @stuartarchibald for the patch!

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,3 +1,242 @@
...
+* ``np.average``
+* ``np.argmin`` gains support for the ``axis`` kwarg.
+* ``np.ndarray.astype`` gains support for types expressed as literal strings.
+
+Highlights of core changes:
+
+* For users of the Numba extension API, Numba now has a new error handling mode
+  whereby it will treat all exceptions that do not inherit from
+  ``numba.errors.NumbaException`` as a "hard error" and immediately unwind the
+  stack. This makes it much easier to debug when writing ``@overload``s etc from
```
Comment (sklam) [inline CHANGE_LOG:34]: the `s` after `@overload` is breaking the format.

**Comment by stuartarchibald** [comment]
> Prose taken from: [stuartarchibald@2d8b802](https://github.com/stuartarchibald/numba/commit/2d8b802b7a6b83407f96d10fa926f5b4f168da2f)
> 
> Thank you @stuartarchibald for the patch!

... and thanks for @gmarkall for providing the `Highlights of changes for the CUDA target:` section.

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,3 +1,242 @@
...
+* PR `#7597 <https://github.com/numba/numba/pull/7597>`_: Update gdb docs for new DWARF enhancements. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7605 <https://github.com/numba/numba/pull/7605>`_: Fix TBB 2021 DSO names on OSX/Win and make TBB reporting consistent (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7610 <https://github.com/numba/numba/pull/7610>`_: Fix #7609. Type should not be mutated. (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#7618 <https://github.com/numba/numba/pull/7618>`_: Fix the doc build: docutils 0.18 not compatible with pinned sphinx (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7627 <https://github.com/numba/numba/pull/7627>`_: PR 7321 continued (`stuartarchibald <https://github.com/stuartarchibald>`_ `Eric Wieser <https://github.com/eric-wieser>`_)
+* PR `#7632 <https://github.com/numba/numba/pull/7632>`_: Capture output in CUDA matmul doctest (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#7637 <https://github.com/numba/numba/pull/7637>`_: Increase the timeout on the SVML tests for loaded machines. (`stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:202]: Reminder to update this as the few final PRs are merged.

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,3 +1,242 @@
...
+  whereby it will treat all exceptions that do not inherit from
+  ``numba.errors.NumbaException`` as a "hard error" and immediately unwind the
+  stack. This makes it much easier to debug when writing ``@overload``\s etc
+  from the extension API as there's now no confusion between Python errors and
+  Numba errors. This feature can be enabled by setting the environment
+  variable: ``NUMBA_CAPTURED_ERRORS='new_style'``.
+* The threading layer selection priority can now be changed via the environment
+  variable ``NUMBA_THREADING_LAYER_PRIORITY``.
+
+Highlights of changes for the CUDA target:
```
Comment (stuartarchibald) [inline CHANGE_LOG:41]: @gmarkall As code owner for CUDA, any chance you could suggest an addition for the recently approved half precision support: https://github.com/numba/numba/pull/7460 ? Many thanks.

**Comment by esc** [comment]
The changelog has been updated.

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,3 +1,257 @@
...
+* PR `#7443 <https://github.com/numba/numba/pull/7443>`_: Refactor template selection logic for targets. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7444 <https://github.com/numba/numba/pull/7444>`_: use correct variable name in closure (`esc <https://github.com/esc>`_)
+* PR `#7447 <https://github.com/numba/numba/pull/7447>`_: cleanup Numba metadata (`esc <https://github.com/esc>`_)
+* PR `#7453 <https://github.com/numba/numba/pull/7453>`_: CUDA: Provide stream in async_done result (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#7456 <https://github.com/numba/numba/pull/7456>`_: Fix invalid codegen for #7451. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7457 <https://github.com/numba/numba/pull/7457>`_: Factor out target registry selection logic (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7459 <https://github.com/numba/numba/pull/7459>`_: Include compiler flags in symbol mangling (`Siu Kwan Lam <https://github.com/sklam>`_)
+* PR `#7460 <https://github.com/numba/numba/pull/7460>`_: Testhound/fp16 support (`Michael Collison <https://github.com/testhound>`_ `Graham Markall <https://github.com/gmarkall>`_)
```
Comment (sklam) [inline CHANGE_LOG:171]: I renamed the PR to be more descriptive
```suggestion
* PR `#7460 <https://github.com/numba/numba/pull/7460>`_: Add FP16 support for CUDA (`Michael Collison <https://github.com/testhound>`_ `Graham Markall <https://github.com/gmarkall>`_)
```

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,3 +1,257 @@
...
+* NumPy version 1.21 is supported.
+* The minimum supported NumPy version is raised to 1.18 for runtime (compilation
+  however remains compatible with NumPy 1.11).
+
+
+Pull-Requests:
+
+* PR `#6075 <https://github.com/numba/numba/pull/6075>`_: add np.float_power and np.cbrt (`Guilherme Leobas <https://github.com/guilhermeleobas>`_)
+* PR `#7047 <https://github.com/numba/numba/pull/7047>`_: Support __hash__ for numpy.datetime64 (`Guilherme Leobas <https://github.com/guilhermeleobas>`_ `stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7057 <https://github.com/numba/numba/pull/7057>`_: Fix #7041: Add charseq registry to CUDA target (`Graham Markall <https://github.com/gmarkall>`_ `stuartarchibald <https://github.com/stuartarchibald>`_)
```
Comment (sklam) [inline CHANGE_LOG:82]: Add 7082
```suggestion
* PR `#7057 <https://github.com/numba/numba/pull/7057>`_: Fix #7041: Add charseq registry to CUDA target (`Graham Markall <https://github.com/gmarkall>`_ `stuartarchibald <https://github.com/stuartarchibald>`_)
* PR `#7082 <https://github.com/numba/numba/pull/7082>`_: Added Add/Sub between datetime64 array and timedelta64 scalar (`Nick Riasanovsky <https://github.com/njriasan>`_ `stuartarchibald <https://github.com/stuartarchibald>`_)
```

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,3 +1,257 @@
...
+  stack. This makes it much easier to debug when writing ``@overload``\s etc
+  from the extension API as there's now no confusion between Python errors and
+  Numba errors. This feature can be enabled by setting the environment
+  variable: ``NUMBA_CAPTURED_ERRORS='new_style'``.
+* The threading layer selection priority can now be changed via the environment
+  variable ``NUMBA_THREADING_LAYER_PRIORITY``.
+
+Highlights of changes for the CUDA target:
+
+* Support for NVIDIA's CUDA Python bindings.
```
Comment (sklam) [inline CHANGE_LOG:43]: Add mentioning of FP16
```suggestion
* Support for NVIDIA's CUDA Python bindings.
* Support for 16-bit floating point numbers and their basic operations via intrinsics.
```

**Review by sklam — COMMENTED** [review]
Suggested updates:

File: CHANGE_LOG (hunk 3 above)
Comment (sklam) [reply CHANGE_LOG:41]: i added in https://github.com/numba/numba/pull/7644/commits/b6bcb18523289b79aaa39d3445ec5eea2274a8f1

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,3 +1,259 @@
+Version 0.55.0 (10 December, 2021)
```
Comment (sklam) [inline CHANGE_LOG:1]:
```suggestion
Version 0.55.0 (15 December, 2021)
```

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,3 +1,253 @@
...
+* PR `#7628 <https://github.com/numba/numba/pull/7628>`_: Move to using windows-2019 images in Azure (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7632 <https://github.com/numba/numba/pull/7632>`_: Capture output in CUDA matmul doctest (`Graham Markall <https://github.com/gmarkall>`_)
+* PR `#7636 <https://github.com/numba/numba/pull/7636>`_: Copy prange loop header to after the parfor. (`Todd A. Anderson <https://github.com/DrTodd13>`_)
+* PR `#7637 <https://github.com/numba/numba/pull/7637>`_: Increase the timeout on the SVML tests for loaded machines. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7645 <https://github.com/numba/numba/pull/7645>`_: In debuginfo, do not add noinline to functions marked alwaysinline (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7650 <https://github.com/numba/numba/pull/7650>`_: Move Azure builds to OSX 10.15 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`
```
Comment (stuartarchibald) [inline CHANGE_LOG:212]: Something has gone wrong with the syntax on this line? Is the end missing?

File: CHANGE_LOG [hunk 9]
```diff
@@ -1,3 +1,253 @@
...
+* PR `#7552 <https://github.com/numba/numba/pull/7552>`_: Prepend the inlined function name to inlined variables. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7557 <https://github.com/numba/numba/pull/7557>`_: Python3.10 v2 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_)
+* PR `#7560 <https://github.com/numba/numba/pull/7560>`_: Refactor with detection py310 (`Siu Kwan Lam <https://github.com/sklam>`_ `esc <https://github.com/esc>`_)
+* PR `#7561 <https://github.com/numba/numba/pull/7561>`_: fix a typo (`Kolen Cheung <https://github.com/ickc>`_)
+* PR `#7567 <https://github.com/numba/numba/pull/7567>`_: Update docs to note meetings are public. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7570 <https://github.com/numba/numba/pull/7570>`_: Update the docs and error message for errors when importing Numba. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#7580 <https://github.com/numba/numba/pull/7580>`_: Fix #7507. catch `NotImplementedError ` in `.get_function()`  (`Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:193]:
```suggestion
* PR `#7580 <https://github.com/numba/numba/pull/7580>`_: Fix #7507. catch `NotImplementedError` in `.get_function()`  (`Siu Kwan Lam <https://github.com/sklam>`_)
```
I think that might make it render correctly?

File: CHANGE_LOG (hunk 8 above)
Comment (sklam) [reply CHANGE_LOG:212]: looks like missing the `_`

Comment (sklam) [reply CHANGE_LOG:212]:
```suggestion
* PR `#7650 <https://github.com/numba/numba/pull/7650>`_: Move Azure builds to OSX 10.15 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>_)`
```

**Comment by esc** [comment]
Screen Shot 2021-12-15 at 19 38 36

The syntax for the final PR in the list is now correct too.

**Review by sklam — APPROVED** [review]
LGTM

**Pull request merged**